### ISSUE TITLE
sh: start-storybook: command not found

### ISSUE THREAD

**Issue by KornbergFresnel** [kadirahq/react-storybook#311]
**what's wrong?**

```
> redux-todomvc-example@0.0.0 storybook /Users/zhouming/Desktop/model/storybookmodel
> start-storybook -p 9001

sh: start-storybook: command not found

npm ERR! Darwin 15.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "storybook"
npm ERR! node v6.2.2
npm ERR! nam  v3.9.5
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! sys call spawn
npm ERR! redux-todomvc-example@0.0.0 storybook: `start-storybook -p 9001`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the redux-todomvc-example@0.0.0 storybook script 'start-storybook -p 9001'edux-todomvc-example@0.0.0 storybook /Users/zhouming/Desktop/model/storybookmodel
> start-storybook -p 9001

sh: start-storybook: command not found

npm ERR! Darwin 15.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "storybook"
npm ERR! node v6.2.2
npm ERR! nam  v3.9.5
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! sys call spawn
npm ERR! redux-todomvc-example@0.0.0 storybook: `start-storybook -p 9001`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the redux-todomvc-example@0.0.0 storybook script 'start-storybook -p 9001'
```

**Comment by KornbergFresnel** [comment]
I've solved it by myself, because I didn't use: 'npm install'

**Comment by arunoda** [comment]
Great.

**Comment by alexthebake** [comment]
I'm also experiencing this issue:
```
→ npm run storybook

> client@0.0.0 storybook /Users/alexbake/espark/frontier-client
> start-storybook -p 6006

sh: start-storybook: command not found

npm ERR! Darwin 16.3.0
npm ERR! argv "/usr/local/Cellar/node/7.4.0/bin/node" "/usr/local/bin/npm" "run" "storybook"
npm ERR! node v7.4.0
npm ERR! npm  v4.0.5
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! client@0.0.0 storybook: `start-storybook -p 6006`
npm ERR! spawn ENOENT
```

```
→ npm -v
4.0.5
```

```
→ node -v
v7.4.0
```

Any ideas?

**Comment by bompi88** [comment]
I did `rm -rf node_modules` and then `npm install`.

**Comment by axed** [comment]
Still the same here on Windows 10, Node 8.0 and NPM 5.0.3

Used Instructions from the website
```
npm i -g @storybook/cli
cd my-react-app
getstorybook
```
Same error.

**Comment by shilman** [comment]
@UsulPro do you mind looking into the Windows situation for `@storybook/cli` and help @axed ?

**Comment by UsulPro** [comment]
```cmd
λ node -v
v8.1.2
```

```cmd
λ npm ls -g --depth 0
e:\Node\nodejs
+-- @storybook/cli@3.2.0-alpha.0
+-- create-react-app@1.3.1
`-- npm@5.0.3
```

```cmd
create-react-app test
cd test
getstorybook
rem yarn add @storbook/components --dev
rem to fix @storbook/components issue
yarn storybook
```

```cmd
yarn storybook v0.19.1
$ start-storybook -p 9009 -s public
@storybook/react v3.1.3

=> Loading static files from: E:\DevelopWeb\_Storybook\sandbox\test1\public .
=> Loading custom addons config.
=> Using default webpack setup based on "Create React App".
webpack built 22ebae935fef959ad46c in 5352ms
Storybook started on => http://localhost:9009/
```

so it installs and works fine. But actually, when I on the Windows I use [cmder](http://cmder.net/) instead of native `cmd.com` - it's the only way to survive with Windows for me! :relaxed:

I tried to do the same on the cmd terminal and I've got an error after `npm run storybook` (`getstorybook` works fine actually):

<summary>

<details>

```cmd
@storybook/react v3.1.3

=> Loading static files from: e:\DevelopWeb\_Storybook\sandbox\test2\public .
=> Loading custom addons config.
=> Using default webpack setup based on "Create React App".
 10% building modules 0/1 modules 1 active ...ok\react\dist\client\manager\index
 10% building modules 1/2 modules 1 active ...ok\sandbox\test2\.storybook\config
 10% building modules 2/2 modules 0 activeevents.js:182
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::9009
    at Object.exports._errnoException (util.js:1016:11)
    at exports._exceptionWithHostPort (util.js:1039:20)
    at Server.setupListenHandle [as _listen2] (net.js:1307:14)
    at listenInCluster (net.js:1355:12)
    at Server.listen (net.js:1455:7)
```
</details>

</summary>
 
but since I didn't do such tests before I can't say is it a new issue or it existed before.

Anyway, I'm not sure that we could still have issues related to `sh: start-storybook` since we don't use shell commands in our scripts. So I guess @axed's case is something new and possible it's worth to submit a new issue starting with detailed reproducing info

PS
@shilman at this stage of @3.2.0-alpha we have an issue with `@storybook/components` requiring

**Comment by shilman** [comment]
Thanks @UsulPro . I'm pretty sure we are going to have to do a lot of surgery on `3.2.0-alpha` before it comes out of alpha. I have not been involved in it much yet.

**Comment by bailinlin** [comment]
I'm also experiencing this issue:

```
 npm run storybook   

> react-redux-saga-boilerplate@0.4.0 storybook /Users/dtx/seahorse-web
> start-storybook -p 6006

sh: start-storybook: command not found
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! react-redux-saga-boilerplate@0.4.0 storybook: `start-storybook -p 6006`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the react-redux-saga-boilerplate@0.4.0 storybook script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/dtx/.npm/_logs/2017-07-08T09_04_22_397Z-debug.log
```

 and there is my error log : 

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'run',
1 verbose cli   'storybook' ]
2 info using npm@5.0.4
3 info using node@v7.0.0
4 verbose run-script [ 'prestorybook', 'storybook', 'poststorybook' ]
5 info lifecycle react-redux-saga-boilerplate@0.4.0~prestorybook: react-redux-saga-boilerplate@0.4.0
6 silly lifecycle react-redux-saga-boilerplate@0.4.0~prestorybook: no script for prestorybook, continuing
7 info lifecycle react-redux-saga-boilerplate@0.4.0~storybook: react-redux-saga-boilerplate@0.4.0
8 verbose lifecycle react-redux-saga-boilerplate@0.4.0~storybook: unsafe-perm in lifecycle true
9 verbose lifecycle react-redux-saga-boilerplate@0.4.0~storybook: PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/Users/dtx/seahorse-web/node_modules/.bin:/Users/dtx/.rvm/gems/ruby-2.3.0/bin:/Users/dtx/.rvm/gems/ruby-2.3.0@global/bin:/Users/dtx/.rvm/rubies/ruby-2.3.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/dtx/.rvm/bin
10 verbose lifecycle react-redux-saga-boilerplate@0.4.0~storybook: CWD: /Users/dtx/seahorse-web
11 silly lifecycle react-redux-saga-boilerplate@0.4.0~storybook: Args: [ '-c', 'start-storybook -p 6006' ]
12 info lifecycle react-redux-saga-boilerplate@0.4.0~storybook: Failed to exec storybook script
13 verbose stack Error: react-redux-saga-boilerplate@0.4.0 storybook: `start-storybook -p 6006`
13 verbose stack spawn ENOENT
13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:33:16)
13 verbose stack     at emitTwo (events.js:106:13)
13 verbose stack     at ChildProcess.emit (events.js:191:7)
13 verbose stack     at maybeClose (internal/child_process.js:877:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
14 verbose pkgid react-redux-saga-boilerplate@0.4.0
15 verbose cwd /Users/dtx/seahorse-web
16 verbose Darwin 15.6.0
17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "storybook"
18 verbose node v7.0.0
19 verbose npm  v5.0.4
20 error file sh
21 error code ELIFECYCLE
22 error errno ENOENT
23 error syscall spawn
24 error react-redux-saga-boilerplate@0.4.0 storybook: `start-storybook -p 6006`
24 error spawn ENOENT
25 error Failed at the react-redux-saga-boilerplate@0.4.0 storybook script.
25 error This is probably not a problem with npm. There is likely additional logging output above.
26 verbose exit [ 1, true ]
```

**Comment by UsulPro** [comment]
@bailinlin could you give us additional information:
- what version of storybook do yo use?
- what version of getstorybook / @storybook/cli do you use?
- what is your OS?

**Comment by axed** [comment]
A lil update from my side. Maybe im understanding the whole installation process wrong, but one simple `npm install whateveruwant` will uninstall nearly all of storybook's dependencies after you installed and setup storybook.
I was a little surprised when i did a simple `npm install apollo-client` and suddenly npm removed like 1138 packages, added 1 package and storybook stopped working.
I have all of my storybook plugins and storybook itself aswell as non-dev dependencies within my `package.json`, to avoid situation like these.

**Comment by UsulPro** [comment]
@axed thanks for reporting this!

could you clarify? do you mean that `npm i smth` removes everything from devDependencies but holds dependencies?

and could you give us info about your OS and NPM version?

**Comment by UsulPro** [comment]
looks like npm bug on Windows

**Comment by axed** [comment]
@UsulPro sure thing!
My `package.json`'s `devDependencies` object is empty, or else it's etries had been removed upon each `npm install somethingnew`.
So i guess yes, every node module that is not listed in your `package.json`'s `dependencies` object/list, will be removed from the local `node_modules` directory.

OS: Windows 10 1608 build 14393.1358
NPM: 5.0.3

**Comment by frankred** [comment]
The problem is that there is no start-storybook cmd or binary for windows. How to get this tool?

**Comment by demiro** [comment]
is this closed? I have the same issue with the latest storybook (@next - 5.2.0-beta.19)

I installed cli globally and I get for example getstorybook functioning... but for some reason the start-storybook, build-storybook commands result in errors:

> build-storybook : The term 'build-storybook' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the...

been trying to make storybook work on my windows machine for past couple of hours... but nothing works (the npx -p @storybook/cli sb init works completely fine)

**Comment by shilman** [comment]
@demiro is it only broken in 5.2.x, or do you know if it's also broken in earlier versions?

**Comment by neginbasiri** [comment]
I have the same issue using yarn run storybook

**Comment by dweng0** [comment]
Seem to be getting this issue with Create react app, have removed the node_modules folder and re-installed. Doesn't seem to fix the issue

**Comment by collinsmwade** [comment]
Seeing this issue as well.

- Storybook 5.1.11
- NPM v6.9.0
- node v10.16.0
- yarn v1.17.3

**Comment by lolmaus** [comment]
Had this issue in an Ember app, this enabled the `start-storybook` command for me:

```
npm i -D @storybook/ember
yarn add -D @storybook/ember
```

**Comment by teunw** [comment]
lolmaus's solution also works for @storybook/angular

**Comment by WretchedDade** [comment]
I had the same issue with @storybook/react and solved it as mentioned above:

```
npm i -D @storybook/react
yarn add -D @storybook/react
```

**Comment by nazreen** [comment]
I had the same issue with @storybook/vue and solved it as mentioned above:

npm i -D @storybook/vue
yarn add -D @storybook/vue

**Comment by Rodnorm** [comment]
> lolmaus's solution also works for @storybook/angular.
> 
> ```
> npm i -D @storybook/angular
> yarn add -D @storybook/angular
> ```

Also worked for react xD

npm i -D @storybook/react && yarn add -D @storybook/react

**Comment by matheo** [comment]
@Rodnorm using both npm dependency managers is not a good idea.
You should stick to one to manage a repo.

**Comment by ImedAdel** [comment]
By the way, if you used the CLI to initialize your project and then maybe tried using Docker and it didn't work, you have to install the CLI as a dependency too.
```sh
# Using npm
npm i -D @storybook/cli

# Using yarn
yarn add -D @storybook/cli
```

**Comment by kasperstorgaard** [comment]
> By the way, if you used the CLI to initialize your project and then maybe tried using Docker and it didn't work, you have to install the CLI as a dependency too.
> 
> ```shell
> # Using npm
> npm i -D @storybook/cli
> 
> # Using yarn
> yarn add -D @storybook/cli
> ```

This change finally made it work for me on netlify, without having to clear cache on every rebuild

**Comment by BiskremMuhammad** [comment]
try to delete `node_modules` and clean nm cache `npm clean cache --force` npm run `npm install` again,

**Comment by rpearce** [comment]
👆 I believe it's `npm cache clean --force`

**Comment by dpedersoli** [comment]
I'm having this issue. Can someone please help me?
![image](https://user-images.githubusercontent.com/63932322/200140108-994d0e72-16ee-4e3d-be33-eee6a5bd37aa.png)

I've already tryied delete "node modules" -> clean cache -> "npm install"

(-npm -v -> 8.10.0

-node -v -> 16.11.1

'"@storybook/react": "^6.5.13"')

**Comment by tkbremnes** [comment]
I've got the same issue, running `yarn storybook` after upgrading to `7.0.0-beta.38` does not work.

What does work is to copy the `package.json` (updated to v7) to a different folder, installing, running `npx sb@next init`. `yarn storybook` will work in this folder.

I don't know enough of how `npm` sets up its binaries and such to know how to troubleshoot this any further. The nuke option by @pawelkrystkiewicz above here has no effect.

**Comment by sbland** [comment]
Also having this issue after upgrading to v7.0.2. 
Package diff below...

```
    "@storybook/addon-actions": "^5.1.9",
    "@storybook/addon-links": "^5.1.9",
    "@storybook/addons": "^5.1.9",
    "@storybook/react": "^5.1.9",
```
to 
```
    "@storybook/addon-actions": "^7.0.2",
    "@storybook/addon-links": "^7.0.2",
    "@storybook/addons": "^7.0.2",
    "@storybook/react": "^7.0.2",
```

**Comment by noru** [comment]
> 



> Also having this issue after upgrading to v7.0.2. Package diff below...
> 
> ```
>     "@storybook/addon-actions": "^5.1.9",
>     "@storybook/addon-links": "^5.1.9",
>     "@storybook/addons": "^5.1.9",
>     "@storybook/react": "^5.1.9",
> ```
> 
> to
> 
> ```
>     "@storybook/addon-actions": "^7.0.2",
>     "@storybook/addon-links": "^7.0.2",
>     "@storybook/addons": "^7.0.2",
>     "@storybook/react": "^7.0.2",
> ```

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#start-storybook--build-storybook-binaries-removed

upgrade script seems didn't pop those warnings. reinstall did.

**Comment by AdamMescher** [comment]
> https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#start-storybook--build-storybook-binaries-removed

From the Storybook Docs:

A typical Storybook project includes two scripts in your projects package.json:

```
{
  "scripts": {
    "storybook": "start-storybook <some flags>",
    "build-storybook": "build-storybook <some flags>"
  }
}
```

To convert this project to 7.0:

```
{
  "scripts": {
    "storybook": "storybook dev <some flags>",
    "build-storybook": "storybook build <some flags>"
  },
  "devDependencies": {
    "storybook": "next"
  }
}
```

tl;dr -- `build-storybook` should be `storybook build` 

I was attempting to run `npx chromatic --project-token=<token>` and that very simple change is the difference.

**Comment by JBustin** [comment]
Hello,

Same error.
I have installed the latest version of @storybook/vue
```
    "@storybook/vue": "7.0.12",
```
and the binaries are now missing (only my test-storybook for test-runner, but not related...):
```sh
$ ls ./node_modules/.bin/*storybook*
./node_modules/.bin/test-storybook
```

@shilman, or other, any idea ?

Regards

**Comment by bvaughn** [comment]
Noticed this myself, using the latest storybook release.

```
$ yarn why storybook  
└─ recordreplay-devtools@workspace:.
   └─ storybook@npm:7.0.24 (via npm:^7.0.24)

$ yarn build-storybook
command not found: build-storybook
```

**Comment by jonniebigodes** [comment]
@bvaughn thanks for reaching out to us. Appreciate it 🙏 ! Something is off I agree with that. And to your point, the documentation is not wrong. The first link you've mentioned (the outdated one), is correct, and it refers to one of the scripts added when Storybook is installed, specifically the [one](https://github.com/chromaui/learnstorybook-code/blob/master/package.json#L27) for generating a production build and it should have been updated if you're migrating or added if it's a fresh install.

**Comment by bvaughn** [comment]
> And to your point, the documentation is not wrong.

Interesting! The docs [here](https://github.com/storybookjs/storybook/issues/311#:~:text=outdated%20documentation) show
```
yarn build-storybook
```

Which, as of Storybook v6, _used_ to be the way you built story book (via the binary added in `node_modules`) but doesn't seem to be the way any more.

The link you sent seems to be a particular project _using_ Storybook:
```json
    "build-storybook": "storybook build",
```

The command it's running is actually `storybook build` (which is also what the [installation page](https://storybook.js.org/docs/react/get-started/install) mentions).

Maybe the docs could be clarified to avoid confusion for folks, given the binary rename between majors?

**Comment by bvaughn** [comment]
The "migration" page you linked to looks fine. It's not the page I ended up on when coming from Google search, which is what caused my confusion.

Normally I'd be happy to open a PR with a proposed docs change. In this case, I'm not sure how well it would fit into the larger tutorial (which often refers to the `package.json` script named `build-storybook` rather than the binary).

I'll defer to your judgement on whether that's worth changing or not.
